### ISSUE TITLE
Handling Overrides

### ISSUE THREAD

**Issue by thombruce** [thombruce/helvellyn#72]
https://edgeguides.rubyonrails.org/engines.html#overriding-models-and-controllers

You'll find there details about overriding models and controllers from the Rails docs.

I had been using `require_dependency` and, well actually see here: https://github.com/thombruce/helvellyn/compare/3bdff8da...0b13a305#diff-48a7c8575f8bf43ed3e369b64a960919

I've rolled back beyond this point to resolve a 503 error on Heroku. This _seems_ to be the offending change. A build with a total class override is in progress, so we should see shortly...

It's a hideous override though, that will become stale against the upstream engine if changes occur. **Absolutely** look at how Rails recommends overrides are added above.

**Comment by thombruce** [comment]
Assumption above was incorrect... but I still feel probably that the controller is to blame. It actually has a second constant in it that would be undefined without the Engine. Maybe the issue is loading order and an uninitialized constant is to blame? _Why are the logs not more helpful?_

Deploying a release now that removes the override controllers entirely. This will remove our capability to allow users to set a name at signup.

**Comment by thombruce** [comment]
**Know that** _this_ is why you check production after every deploy, and use review apps!

**Comment by thombruce** [comment]
Possible breaking changes between working version and latest:

Moved out of ApplicationController into scopes:

```
after_action :verify_authorized
after_action :verify_policy_scoped, only: :index
```

Name change of included module:
```
include Credible::ControllerConcern
```

Added to ApplicationMailer:
```
before_action { params[:app_name] = Settings.instance.name }
```

```
default from: -> { Settings.instance.email || 'helvellyn@thombruce.com' }
```

**Comment by thombruce** [comment]
Removal of Credible::Authentication::UserPolicy seems to have done the trick...

...must've been failing to load due to lack of dependencies, as this was removed from Credible.

...

Just need to restore UsersController override to get back on track.

**Comment by thombruce** [comment]
To achieve that, let's go with the ActiveSupport::Concern approach and modularize our Engine logic that way, over on https://github.com/thombruce/credible

**Comment by thombruce** [comment]
closed by 948f12aa6d729c0e8f02a712ee27ccdd68033352

We now handle controller logic as concerns in Credible, enabling easy extension.
